contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1916,6 +1920,186 @@ checksum(PG_FUNCTION_ARGS)
+	
+	/*
+	 * Get metadata of partition function for the provided partition
+	 * function name, if it exists in the provided database.
+	 */
+	rel = table_open(get_bbf_partition_function_oid(), AccessShareLock);
+	
+	ScanKeyInit(&scanKey[0],
+				Anum_bbf_partition_function_dbid,
+				BTEqualStrategyNumber, F_INT2EQ,
+				Int16GetDatum(dbid));
+
+	ScanKeyEntryInitialize(&scanKey[1], 0,
+				Anum_bbf_partition_function_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(partition_func_name));
+	
+	/* Scan using index. */
+	scan = systable_beginscan(rel, get_bbf_partition_function_pk_idx_oid(),
+			false, NULL, 2, scanKey);
+	
+	tuple = systable_getnext(scan);
+	if (HeapTupleIsValid(tuple))
+	{
+		func_param_typname = TextDatumGetCString(heap_getattr(tuple, Anum_bbf_partition_function_input_parameter_type, RelationGetDescr(rel), &isnull));
+		values = DatumGetArrayTypeP(heap_getattr(tuple, Anum_bbf_partition_function_range_values, RelationGetDescr(rel), &isnull));
+		deconstruct_array(values, sqlvariant_typoid,
+					-1, false, 'i', &range_values, &nulls, &nelems);
+	}
+	systable_endscan(scan);
+	table_close(rel, AccessShareLock);
+
+	/* Raise error if provided partition function doesn't exist in the provided database. */
+	if (!func_param_typname)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("Invalid object name '%s'.", partition_func_name)));
+	
+	/* NULL values will always belong to first partition. */
+	if (PG_ARGISNULL(1))
+	{
+		pfree(partition_func_name);
+		pfree(func_param_typname);
+		pfree(nulls);
+		pfree(range_values);
+		PG_RETURN_INT32(1);
+	}
+
+	argtypeid = get_fn_expr_argtype(fcinfo->flinfo, 1);
+	arg = PG_GETARG_DATUM(1);
+
+	/* Get OID of partition function parameter type. */
+	func_param_typoid = (*common_utility_plugin_ptr->get_tsql_datatype_oid) (func_param_typname);
+
+	/* 
+	 * Implicitly convert input value to parameter type of
+	 * partition function and it will raise error if conversion fails.
+	 */
+	arg = pltsql_exec_tsql_cast_value(arg, &isnull, argtypeid, -1, func_param_typoid, -1);
+
+	/* Cast it to sql_variant */
+	arg = pltsql_exec_tsql_cast_value(arg, &isnull,
+						func_param_typoid, -1,
+						sqlvariant_typoid, -1);
+
+	/*
+	 * Find oid of comparator function for input type, which will be used during the sorting.
+	 * Here, we are first finding the default operator class for the input type then using that
+	 * we are finding the operator family for that operator class and finally using that we are
+	 * finding the defined comparator function for that operator family.
+	 */
+	basetype_oid = getBaseType(sqlvariant_typoid);
+	opclass_oid = GetDefaultOpClass(basetype_oid, BTREE_AM_OID);
+	opfamily_oid = get_opclass_family(opclass_oid);
+	cmpfunction_oid = get_opfamily_proc(opfamily_oid, basetype_oid, basetype_oid,
+						BTORDER_PROC);
```

Comment:
Lets use the sql_variant_cmp oid directly

Reply:
addressed.